src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +325,98 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        private readonly List<WeakReference<Stats>?>? _allStats;
+        private readonly Stats? _accumulatedStats;
+        private ThreadLocal<Stats>? _stats;
+
+        private (long, long) Sum()
+        {
+            long hits = 0, misses = 0;
+            lock (_allStats!)
+            {
+                hits += _accumulatedStats!.Hits;
+                misses += _accumulatedStats!.Misses;
+                foreach (WeakReference<Stats>? wr in _allStats!)
+                {
+                    if (wr is not null && wr.TryGetTarget(out Stats? stats))
+                    {
+                        hits += Interlocked.Read(ref stats.Hits);
+                        misses += Interlocked.Read(ref stats.Misses);
+                    }
+                }
+
+                _allStats.TrimExcess();
```

Comment:
Why TrimExcess during Sum()? Wouldn't it be more appropriate to do it during RemoveFromStats?

My reasoning is if no one calls GetCurrentStatistics(), the excess won't be trimmed. And also, you won't need to trim unless you removed something from the list - and typically, I can imagine removing will happen less often than calling GetCurrentStatistics().